integration-tests/tests/src/tests/sync/mixed.ts

```diff
@@ -0,0 +1,134 @@
+import { assert, expect } from "chai";
+
+import { importAppBefore, authenticateUserBefore, openRealmBefore } from "../../hooks";
+
+import { itUploadsDeletesAndDownloads } from "./upload-delete-download";
+
+type MixedClass = { _id: Realm.BSON.ObjectId, value: Realm.Mixed, list: Realm.List<Realm.Mixed> };
+type Value = Realm.Mixed | ((realm: Realm) => Realm.Mixed);
+type ValueTester = (actual: Realm.Mixed, inserted: Realm.Mixed) => void | boolean;
+
+/**
+ * The default tester of values.
+ * @param actual The value downloaded from the server.
+ * @param inserted The value inserted locally before upload.
+ */
+function defaultTester(actual: Realm.Mixed, inserted: Realm.Mixed) {
+  expect(actual).equals(inserted);
+}
+
+/**
+ * Registers a test suite that:
+ * - Opens a synced Realm
+ * - Performs an object creation
+ * - Uploads the local Realm
+ * - Deletes the Realm locally
+ * - Reopens and downloads the Realm
+ * - Performs a test to ensure the downloaded value match the value created locally.
+ * @param typeName 
+ * @param options 
+ */
+function describeRoundtrip(typeName: string, value: Value, testValue: ValueTester = defaultTester) {
+  function performTest(actual: Realm.Mixed, inserted: Realm.Mixed) {
+    const result = testValue(actual, inserted);
+    if (typeof result === "boolean") {
+      expect(result).equals(true, `${testValue} failed!`);
+    }
+  }
+
+  describe(`roundtrip of '${typeName}'`, () => {
+    openRealmBefore({
+      schema: [{
+        name: "MixedClass",
+        primaryKey: "_id",
+        properties: {
+          _id: "objectId",
+          value: "mixed?",
+          list: "mixed[]",
+        },
+      }],
+      sync: { partitionValue: "mixed-test" },
+    });
+
+    it("writes", function(this: RealmContext) {
+      this._id = new Realm.BSON.ObjectId();
+      this.realm.write(() => {
+        this.value = typeof value === "function" ? value(this.realm) : value;
+        this.realm.create<MixedClass>("MixedClass", {
+          _id: this._id,
+          value: this.value,
+          // Adding a few other unrelated elements to the list
+          list: [this.value, 123, false, "something-else"],
+        });
+      });
+    });
+  
+    itUploadsDeletesAndDownloads();
```

Comment:
I assume that the db state is persisted when this test is called?  Is there cleanup required here?

Reply:
> I assume that the db state is persisted when this test is called?
Depends on what you mean by "db state is persisted".
The test awaits the database being uploaded before deleting it.

> Is there cleanup required here?
Nope, that's done by the `after` hook in `openRealmBefore`.